key_management/src/keystore.rs

```diff
@@ -226,11 +226,17 @@ impl PersistentKeyStore {
             .parent()
             .ok_or_else(|| Error::Other("Invalid Path".to_string()))?;
         fs::create_dir_all(dir)?;
-
         let file = File::create(&self.location)?;
         let writer = BufWriter::new(file);
-        serde_json::to_writer(writer, &self.key_info)
-            .map_err(|e| Error::Other(format!("failed to serialize and write key info: {}", e)))?;
+        if let true = self.is_encrypted {
+            serde_cbor::to_writer(writer, &self.key_info).map_err(|e| {
+                Error::Other(format!("failed to serialize and write key info: {}", e))
+            })?;
+        } else {
+            serde_json::to_writer(writer, &self.key_info).map_err(|e| {
+                Error::Other(format!("failed to serialize and write key info: {}", e))
+            })?;
```

Comment:
Perfect!

Reply:
Well, maybe not perfect if Eric thinks we should just keep track of one file and encrypt the keys

Reply:
I'm not sure he understands the context. Let's keep it this way for now.

Reply:
:+1: